commit 4a28fe6ad69f7b76b3e845264b0afe03473eb6d8
Author: Pierre Boutillier <pierre.boutillier@ens-lyon.org>
Date:   Tue Apr 9 10:18:49 2019 +0200

    `ocamlopt -config` returns "target" `os_type` (instead of "host" one)

diff --git a/Changes b/Changes
index f855bb3f5..b5da51514 100644
--- a/Changes
+++ b/Changes
@@ -52,6 +52,10 @@ Working version
   of an extensible variant type
   (Guillaume Bury, review by many fine eyes)
 
+- Fix `ocamlopt -config` `os_type` entry to return always the
+  "target" os_type
+  (Pierre Boutillier)
+
 ### Compiler distribution build system:
 
 - #2267: merge generation of header programs, also fixing parallel build on
diff --git a/Makefile.config.in b/Makefile.config.in
index 3485f9e1b..fffdf5874 100644
--- a/Makefile.config.in
+++ b/Makefile.config.in
@@ -136,6 +136,8 @@ MODEL=@model@
 
 ### Name of operating system family for the native-code compiler.
 SYSTEM=@system@
+### OCaml name of operating system family for the native-code compiler.
+OCAML_OS_TYPE=@ostype@
 
 ### Command and flags to use for assembling ocamlopt-generated code
 ASM=@AS@
diff --git a/configure b/configure
index c0f2e7296..a749188ea 100755
Binary files a/configure and b/configure differ
diff --git a/configure.ac b/configure.ac
index d9bda3c7e..7b01c2a06 100644
--- a/configure.ac
+++ b/configure.ac
@@ -84,6 +84,7 @@ AC_SUBST([arch])
 AC_SUBST([arch64])
 AC_SUBST([model])
 AC_SUBST([system])
+AC_SUBST([ostype])
 AC_SUBST([systhread_support])
 AC_SUBST([unix_or_win32])
 AC_SUBST([unixlib])
diff --git a/utils/Makefile b/utils/Makefile
index 687529b39..ce6b9e1b5 100644
--- a/utils/Makefile
+++ b/utils/Makefile
@@ -81,6 +81,7 @@ config.ml: config.mlp $(ROOTDIR)/Makefile.config Makefile
 	    $(call SUBST,SYSTEM) \
 	    $(call SUBST,SYSTHREAD_SUPPORT) \
 	    $(call SUBST,TARGET) \
+	    $(call SUBST,OCAML_OS_TYPE) \
 	    $(call SUBST,WITH_FRAME_POINTERS) \
 	    $(call SUBST,WITH_PROFINFO) \
 	    $(call SUBST,WITH_SPACETIME) \
diff --git a/utils/config.mlp b/utils/config.mlp
index 69c561257..7a275ad0c 100644
--- a/utils/config.mlp
+++ b/utils/config.mlp
@@ -185,7 +185,7 @@ let configuration_variables =
   p "ext_asm" ext_asm;
   p "ext_lib" ext_lib;
   p "ext_dll" ext_dll;
-  p "os_type" Sys.os_type;
+  p "os_type" "%%OCAML_OS_TYPE%%";
   p "default_executable_name" default_executable_name;
   p_bool "systhread_supported" systhread_supported;
   p "host" host;
